### PR TITLE
parser: use yyLex.Error instead of (*lex).err

### PR DESCRIPTION
**Pull request by tiancaiamao**
clean up...
See https://github.com/pingcap/tidb/issues/1490

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -1749,7 +1750,7 @@ PredicateExpr:
 	{
 		escape := $5.(string)
 		if len(escape) > 1 {
-			yylex.(*lexer).errf("Incorrect arguments %s to ESCAPE", escape)
+			yylex.Error(fmt.Sprintf("Incorrect arguments %s to ESCAPE", escape))
```
Comment (coocood) [inline parser/parser.y]: Why not add `Errorf` method?

**Comment by zimulala** [comment]
using master branch:
`mysql> create table t(a);
ERROR 1105 (HY000): line 1 column 17 near ")"
`
using your branch:
`mysql> create table t(a);
ERROR 1105 (HY000): line 1 column 17 near ")"expected Types or one of [BIGINT, BINARY, BIT, BLOB, BOOL, BOOLEAN, CHAR, DATE, DATETIME, DECIMAL, DOUBLE, ENUM, INT, INTEGER, LONGBLOB, LONGTEXT, MEDIUMBLOB, MEDIUMINT, MEDIUMTEXT, NATIONAL, NUMERIC, REAL, SET, SMALLINT, TEXT, TIME, TIMESTAMP, TINYBLOB, TINYINT, TINYTEXT, VARBINARY, VARCHAR, YEAR, float, float32, float64, int64, string, uint, uint64]
`

File: parser/goyacc/main.go [hunk 2]
```diff
@@ -641,7 +641,8 @@ yynewstate:
 			if !ok || msg == "" {
 				msg = "syntax error"
 			}
-			yylex.Error(msg)
+			// ignore goyacc error message
```
Comment (tiancaiamao) [inline parser/goyacc/main.go]: previously I neglect a detail that errors generated by goyacc tool are ignored.
now this is fixed @zimulala

**Comment by zimulala** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**